### PR TITLE
feat: Airflow Data Pipeline Construct 

### LINKED ISSUE #109: (core) Airflow (MWWA) data pipeline stage
**Is your feature request related to a problem? Please describe.**
DDK Airflow integration

**Describe the solution you'd like**
I'd like to be able to kickoff MWWA DAG's in DDK DataPipelines. A stage should kickoff the DAG and wait for it to complete.

**Describe alternatives you've considered**
N/A
**Additional context**
N/A

*P.S. Please Don't attach files. Add code snippets directly in the message body instead.*

### PR DESCRIPTION
**Pull request by malachi-constant**
Closes #109

### Features
- Adds basic pipeline construct to get started with airflow -> `AirflowDataPipeline()`
- Adds test and integration test

## Enhancements 
- Sqs -Lambda Stage test coverage increase.

### Discussion
- The intention behind this construct is to accelerate airflow pipeline build for our ddk users. 
- Stage to trigger dags to be included in a separate PR.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: e4172c7c50622aee60d5d896843b02421bb79dd6
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0928ca16-f273-4bfc-8d25-70d4e8a34125%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 35a33a3ddfe839c80e3e070f5d389c7516732c8f
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7dcdd563-7705-49cd-9da5-bb547924739c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 8c9e3dbf6c0286f06d4743152432483398d18a3d
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=958fbead-58a0-4721-8c3b-ae1b42f18a07%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/pipelines/airflow-pipeline.ts
Comment (anmolsgandhi) [inline src/pipelines/airflow-pipeline.ts:1]: can we keep an escape hatch for other MWAA attributes?

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: dbc74d892b6a4bee2e28c7e1e0f002be4cc9ac87
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9a2a80eb-a8b2-4f82-8000-04848762e515%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/pipelines/airflow-pipeline.ts
Comment (malachi-constant) [reply src/pipelines/airflow-pipeline.ts:1]: [85eb76a](https://github.com/awslabs/aws-ddk/pull/349/commits/85eb76a72d6a7cc965e64972fa039fba1364fa69)

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 85eb76a72d6a7cc965e64972fa039fba1364fa69
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ae22bf88-8740-410e-9dfc-7b47ae58f87b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 522657644916328bd63cec94722c03d41d217ada
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c74b5576-afe5-484c-b2d5-9686b6d35381%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — COMMENTED** [review]
Does this construct fit better with core/factories e.g. MWAA factory? It does not extend pipeline/stage interface. 

Also would be great to have MWAA stage that provides runtime capability e.g. kicking off a DAG (if that is achievable) at some point, not necessarily in this PR though

**Comment by malachi-constant** [comment]
> Does this construct fit better with core/factories e.g. MWAA factory? It does not extend pipeline/stage interface.
> 
> Also would be great to have MWAA stage that provides runtime capability e.g. kicking off a DAG (if that is achievable) at some point

@kukushking Yea it definitely is more of a core construct, we can definitely keep it in that subdir for organization's sake. (As we don't use submodules anymore) 

And yea the stage is the next step, but I'm going to address that in a separate PR.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 0197cd0c842e07659b119c6831df5f67138c5ecb
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9dafef8e-ec20-4b6e-8e95-8fd990e92ec5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 1f92b9762e618775bdd5377ca42a0dd91d94f40d
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6dbc8389-ade9-43de-8770-00a0fe0f4e03%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by anmolsgandhi — APPROVED** [review]
1. I see a sqs-lambda-stage.test.ts, making sure if its inteneded in this PR ?

**Comment by malachi-constant** [comment]
> 1. I see a sqs-lambda-stage.test.ts, making sure if its inteneded in this PR ?

Left over from a coverage increase, let's call it a bonus i'll update the description.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 1f92b9762e618775bdd5377ca42a0dd91d94f40d
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c777337e-9bca-473c-a774-7dc6da1cbe04%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 1f92b9762e618775bdd5377ca42a0dd91d94f40d
* Result: SUCCEEDED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1fc8c3bb-d25a-413a-bf69-c1ccc07c89a7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**